Framework/DataHandling/src/SetSampleMaterial.cpp

```diff
@@ -308,8 +324,14 @@ void SetSampleMaterial::exec() {
       // ...but only calculate it if you have both numbers
       if ((!isEmpty(zParameter)) && (!isEmpty(unitCellVolume)))
         rho = zParameter / unitCellVolume;
+      // or from the relative molecular mass if the mass density is specified
+      else if (!isEmpty(rho_m)) {
+        rho = (rho_m / atom.mass) * N_A / 1e24; // measured density in g/cm^3
+      }
     }
-    mat.reset(new Material(chemicalSymbol, neutron, rho));
+    char symbolname[20];
+    sprintf(symbolname, "(%s%i)", atom.symbol.c_str(), a_number);
```

Comment:
Here you could use `stringstream` or other alternaives to make this more C++ish.
`snprintf` would be safer than `sprintf`. I wonder if coverity will complain about `sprintf`.
I don't think this is an issue, but maybe we should discourage old-char-style calls for forward-looking consistency.